src/ethereum/index.ts

```diff
@@ -0,0 +1,871 @@
+        data: this.contractAccounting.methods
+          .claimWithdrawRequest()
+          .encodeABI(),
+      };
+    } catch (error) {
+      throw this.handleError('CLAIM_WITHDRAW_REQUEST_ERROR', error);
+    }
+  }
+
+  /**
+   * Returns the number of validators expected to stop.
+   *
+   * @returns A Promise that resolves to the number of validators expected to stop.
+   *
+   * @throws Will throw an Error if the contract call fails.
+   */
+  public async closeValidatorsStat(): Promise<number> {
+    try {
+      const result = await this.contractAccounting.methods
+        .closeValidatorsStat()
+        .call();
+
+      return Number(result);
+    } catch (error) {
+      throw this.handleError('CLOSE_VALIDATORS_STAT_ERROR', error);
+    }
+  }
+
+  /**
+   * Stakes funds into pool.
+   *
+   * @param address - Sender address.
+   * @param amount - Stake amount in ETH.
+   * @param source - Stake source. Default is '0'.
+   *
+   * @returns A Promise that resolves to the unsigned ETH transaction object.
+   *
+   * @throws Will throw an Error if the amount is not a string, the amount is less than the minimum, or the contract call fails.
+   */
+
+  public async stake(
+    address: string,
+    amount: string,
+    source: string = '0',
+  ): Promise<Transaction> {
+    if (typeof amount !== 'string') {
+      this.throwError('WRONG_TYPE_MESSAGE');
+    }
+
+    const amountWei = this.web3.utils.toWei(amount, 'ether');
+
+    if (new BigNumber(amountWei).lt(this.minAmount)) {
+      this.throwError('MIN_AMOUNT_ERROR', this.minAmount.toString());
+    }
+
+    try {
+      const gasConsumption = await this.contractPool.methods
+        .stake(source)
+        .estimateGas({ from: address, value: amountWei });
+
+      // Create the transaction
+      return {
+        from: address,
+        to: this.addressContractPool,
+        value: Number(amountWei),
+        gasLimit: this.calculateGasLimit(gasConsumption),
+        data: this.contractPool.methods.stake(source).encodeABI(),
+      };
+    } catch (error) {
+      throw this.handleError('STAKE_ERROR', error);
+    }
+  }
+
+  /**
+   * Unstake value from active autocompound balance.
+   * AllowedInterchangeNum is max allowed number interchanges with pending stakers.
+   * Unstaked immediately if value <= pool pending balance or create withdraw request.
+   * Interchange disallowed as default.
+   *
+   * @param address - Sender address.
+   * @param amount - Unstake amount in ETH.
+   * @param allowedInterchangeNum - Max allowed number of interchanges. Default is 0.
+   * @param source - Unstake source. Default is '0'.
+   *
+   * @returns A Promise that resolves to the unsigned ETH transaction object.
+   *
+   * @throws Will throw an Error if the amount is not a string, the balance is less than the amount, or the contract call fails.
+   */
+  public async unstake(
+    address: string,
+    amount: string,
+    allowedInterchangeNum: number = 0,
+    source: string = '0',
```

Comment:
AFAIR, the source param defines what partner uses the SDK. In such a case, it shouldn't have any default values and must a be required param. Also, it's better to have a self-explanatory type for this, instead of string numbers.